commit d3b0f009da0c56490efe58279a7656a613a36b6a
Author: dann frazier <dannf@hp.com>
Date:   Mon Feb 15 21:11:56 2010 -0700

    [Adjusted to apply to Debian's 2.6.18]
    commit f60311d5f7670d9539b424e4ed8b5c0872fc9e83
    Author: Anand V. Avati <avati@gluster.com>
    Date:   Thu Oct 22 06:24:52 2009 -0700
    
        fuse: prevent fuse_put_request on invalid pointer

diff --git a/fs/fuse/file.c b/fs/fuse/file.c
index 4e174c8..b39ddee 100644
--- a/fs/fuse/file.c
+++ b/fs/fuse/file.c
@@ -586,7 +586,8 @@ static ssize_t fuse_direct_io(struct file *file, const char __user *buf,
 				break;
 		}
 	}
-	fuse_put_request(fc, req);
+	if (!IS_ERR(req))
+		fuse_put_request(fc, req);
 	if (res > 0) {
 		if (write) {
 			spin_lock(&fc->lock);
